Problem: not updated to defi-wallet-core v0.3.1 (fix #297)

leejw51crypto opened the pull request:
Solution: update to latest

tomtau commented on .github/workflows/cpp-lint.yml:
```diff
@@ -9,6 +9,10 @@ on:
 jobs:
   cpp-linter:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        compiler: [clang++-14]
```
it seems this matrix/environment isn't being used? the compiler is always hardcoded to 14 elsewhere?

damoncro commented:
Seems we are fixing the same issue: https://github.com/cronos-labs/play-cpp-sdk/pull/292?

damoncro commented:
But as I mentioned on https://github.com/cronos-labs/play-cpp-sdk/pull/292, `.github/workflows/cpp-lint.yml` should be removed, because the linter checked should be conducted after build.

damoncro commented:
You can check the failure details inside https://github.com/cronos-labs/play-cpp-sdk/actions/runs/3635125599/jobs/6133931849

damoncro commented:
I deleted cpp-linter.yml on https://github.com/cronos-labs/play-cpp-sdk/pull/292 to avoid linter issues (it is incorrect setup as I mentioned), please help review and merge before fixing the linter issue. So that we would be break the build because the linter issue.

elecming reviewed (changes requested):
Please check https://github.com/cronos-labs/play-cpp-sdk/pull/292 and merge it first.

leejw51crypto commented:
ok

tomtau commented:
maybe tag defi-wallet-core v0.3.1 first?

leejw51crypto commented:
yes, i'm checking now

leejw51crypto committed:
Problem: incorrect variable name in wallet-connect example (fix #299) (#300)

Solution: use correct name

Problem: pre-stable ethers (#301)

Solution: updated ethers to 1.0 and defi wallet core submodule to the latest one

update

update version

update changelog

update Cargo

change defi-wallet-core version

update changelog

leejw51crypto commented:
rebasing now

leejw51crypto commented:
rebased

damoncro commented:
Please update https://github.com/cronos-labs/play-cpp-sdk/blob/main/play-cpp-sdk/Cargo.toml#L10 as well.

damoncro commented on play-cpp-sdk/Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "play-cpp-sdk"
-version = "0.0.11-alpha"
+version = "0.0.12-alpha"
```
Still missing
`defi-wallet-core-cpp = { path = "../defi-wallet-core-rs/bindings/cpp/", version = "0.3.1"}`
or
`defi-wallet-core-cpp = { path = "../defi-wallet-core-rs/bindings/cpp/", version = "0.3"}`
?

leejw51crypto replied:
ok

damoncro commented on CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
+
+## [v0.0.12-alpha] - 2022-12-7
+- Dynamic Contract C++ Bindings : Encode,Call,Send
+- Minting C++ Example : Encode,Send
```
Changes on play-cpp-sdk itself are needed to add:
- Replace custom cxx-build with official cxx-build
- Added the walletconnect registry API support
- Add app token for sepgrep CI
- Fix Walletconnect request deadlock issue
- Upgrade ethers to 1.0

leejw51crypto commented:
update changelog
